src/charm.py

```diff
@@ -138,6 +141,27 @@ def _on_reset_instance_action(self, event: ActionEvent) -> None:
         event.set_results(results)  # type: ignore[arg-type]
         self.model.unit.status = ops.ActiveStatus()
 
+    def _on_register_user_action(self, event: ActionEvent) -> None:
+        """Reset instance and report action result.
+
+        Args:
+            event: Event triggering the reset instance action.
+        """
+        results = {
+            "register-user": False,
+        }
+        username = event.params["username"]
+        password = event.params["password"]
+        admin = bool(event.params["admin"] == "yes")
+        try:
+            self._synapse_api.register_user(username=username, password=password, admin=admin)
+            results["register-user"] = True
+        except (RegisterUserError, NetworkError) as exc:
+            event.fail(str(exc))
+            return
+        # results is a dict and set_results expects _SerializedData
```

Comment:
If it expects something else, why are we passing `dict`?

Reply:
@yhaliaw can answer that better than me :)
See: https://github.com/canonical/github-runner-operator/pull/26#discussion_r1049138401
I could do as he did and change it to a Dict[str, "JsonObject"] but in this case, dict worked fine.

Reply:
@benhoyt any comment please?

Reply:
in general, using `type: ignore` isn't great so it would be better not to use it

Reply:
I think you just need to update your version of ops. I improve a bunch of type annotations in 2.4.0, so the type of `set_results` is now `set_results(self, results: Dict[str, Any])` instead of using the (private) `_SerializedData`. So you should be able to update ops (I'd use `ops >= 2.4.1`) and then remove the `type: ignore` and it'll work fine.

Reply:
Thanks!